### ISSUE TITLE
Migrate Shipping > Carriers page

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#10585]
Part of Symfony migration project
Type: page

## Status

Todo:
- [x] Migrate listing
- [x] Add helper card
- [ ] Fix old bug #11388
- [x] [Add a modal like on product list (Catalog > Products) to confirm or cancel delete in bulk action](https://github.com/PrestaShop/PrestaShop/issues/14462)
- [x] [Add a modal to confirm or cancel delete action for one row](https://github.com/PrestaShop/PrestaShop/issues/17082)
- [ ] Multistore header: [Specs](https://github.com/PrestaShop/prestashop-specs/blob/7e820f30347fe69ccd3db5a99ab02ef6f7bbdd54/back-office/multistoregeneralspecs.md#multistore-header-only-from-178)

#### Controller / routing
- [ ] All controller actions are migrated and protected with `@AdminSecurity` annotations when eligible and `@DemoRestricted` annotations when eligible
- [ ] All links targeting this page are updated: either using _legacy_link or updating the code
- [ ] The new SF routes have `_legacy_link` and `_legacy_parameters` provided and explained
- [ ] Controller catches Domain Exception and translate them nicely
- [ ] Add/Edit controllers must catch global Exception in order to handle ModuleErrorException
#### Template
- [ ] Twig templates must contain relevant blocks to allow partial extension by modules
- [ ] Twig forms must have `form_rest()` and `form_errors()`
- [ ] JS being used on the page can be used as enabled extensions and use a PageMap for selectors
- [ ] Use `@PrestaShop` instead of `PrestaShopBundle` for template root path aliasing
##### Conventions
- [ ] The new controller actions comply with our naming convention
- [ ] The new SF routes names and paths comply with our naming convention
- [ ] The new Twig template names comply actions with our naming convention
- [ ] Created files (controller, JS assets, CSS assets and twig views) must be in the right folder following the convention
- [ ] CSS being added must follow theme conventions

#### Core layer
- [ ] Commands and Queries only use native types but internally use VOs
- [ ] All Handlers must have an Interface
##### Conventions
- [ ] The created SF service names comply with our naming convention
- [ ] The created files must follow the folder convention (Domain, Core, Adapter...)

#### Polishing
- [ ] The legacy controller and templates files are gone
- [ ] If PR includes a new concept/mechanism, documentation must be written
- [ ] JS and CSS assets must be compiled and up-to-date
- [ ] License headers must be up-to-date
- [ ] New hooks have been added to the hooks list and the SQL updates
- [ ] Performance must be correct: page loading time, filtering/sorting/searching response time, actions response time

#### Testing
- [ ] Commands and Queries are tested with Behat tests
- [ ] Eligible Core classes are covered with unit tests*
- [ ] Controller is covered with survival test
- [ ] (soon) Controller is covered with E2E tests using puppeeter
- [ ] If we break E2E tests because IDs change, we must update the E2E tests

*Eligible = easy to isolate and unit tests provide values (for example testing getters/setters is useless)

**Comment by marionf** [comment]
Issue on Carrier page: #11388
If it can be solved during page migration it will be good, otherwise we will do it later

**Comment by colinegin** [comment]
This page will need a helper card - design needs to be provided @TristanLDD

**Comment by TristanLDD** [comment]
> This page will need a helper card - design needs to be provided @TristanLDD

The helper card design is now ready for integration ;)

**Comment by ks129** [comment]
I try to migrate this page.

**Comment by matks** [comment]
> I try to migrate this page.

Hi ! it's nice to see your motivation but I warn you: because of the heavy and complex JS behavior, this page is very complex!

We have started working on it and when we saw how much work was needed we put it on hold to focus on other topics. We are not 100% sure yet how to structure the JS part.

**Comment by Matt75** [comment]
Is there a plan to manage the high amount of input fields ?
Like an ajax submission range by range instead of submit all fields at the end ?

With standard PHP configuration, max_input_vars is reached when you have many ranges of price/weight and zones...

This cause a lot of issues and many complain on PrestaShop Forums : https://www.prestashop.com/forums/tags/max_input_vars/ (Many topics does not have this tag, you should do a search)

Same problems in many pages of BO, like payment preferences, translations, product page...

**Comment by ks129** [comment]
This goes so complex yes, so maybe it's not the best idea to do this myself. I try to find something easier.

**Comment by matks** [comment]
> This goes so complex yes, so maybe it's not the best idea to do this myself. I try to find something easier.

I would suggest trying "Shop parameters > Customer settings > Titles" (https://github.com/PrestaShop/PrestaShop/issues/10628) or "Customers > Outstanding" (https://github.com/PrestaShop/PrestaShop/issues/17156)

And reading https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/ :)

**Comment by hibatallahAouadni** [comment]
Hello @matks 

Could you please specify this issue status, is it **`To Do`** or **`WIP`**? 
For the moment, I add **`TBS`** label, but please change the status, Thanks!

Ping @marionf this issue is a **Feature** or an **Improvement**? cause it seems it has a lot of tasks :confused: 
Thanks in advance!

**Comment by khouloudbelguith** [comment]
Hi @matks, @marionf,

This issue https://github.com/PrestaShop/PrestaShop/issues/11388 cannot be fixed in this PR: https://github.com/PrestaShop/PrestaShop/pull/20737 because it is blocked by this issue: https://github.com/PrestaShopCorp/ps_mbo/pull/105

Thank you!
